src/client/ui/EditorContainer.js

```diff
@@ -451,35 +437,91 @@ class EditorContainer extends Component {
       await action(uri);
       this.hideDialog();
     } catch (e) {
+      if (e.name === "ConflictError") {
+        this.onSceneErrorOccurred(e, uri, reload);
+      } else {
+        this.setState({
+          openModal: {
+            component: OptionDialog,
+            shouldCloseOnOverlayClick: false,
+            props: {
+              title: "Error Opening Scene",
+              message: `
+                ${e.message}:
+                ${e.url}.
+                Please make sure the file exists and then press "Resolved" to reload the scene.
+              `,
+              options: [
+                {
+                  label: "Resolved",
+                  onClick: () => {
+                    this.setState({ openModal: null });
+                    reload(uri);
+                  }
+                }
+              ],
+              cancelLabel: "Cancel",
+              hideDialog: () => {
+                this.setState({ openModal: null });
+              }
+            }
+          }
+        });
+      }
+    } finally {
+      opened = true;
+    }
+  };
+
+  onSceneErrorOccurred = (error, uri, reload) => {
+    if (error.type === "import") {
+      // empty/duplicate node names in the importing file
       this.setState({
         openModal: {
           component: OptionDialog,
           shouldCloseOnOverlayClick: false,
           props: {
-            title: "Error Opening Scene",
-            message: `
-              ${e.message}:
-              ${e.url}.
-              Please make sure the file exists and then press "Resolved" to reload the scene.
-            `,
+            title: "Resolve Node Conflicts",
+            message:
+              "We've found duplicate and/or missing node names in this file.\nWould you like to fix all conflicts?\n*This will modify the original file.",
             options: [
               {
-                label: "Resolved",
+                label: "okay",
                 onClick: () => {
                   this.setState({ openModal: null });
-                  reload(uri);
+                  this._overwriteConflictsInSource(error.uri, error.handler, () => {
+                    reload(uri);
+                  });
                 }
               }
             ],
-            cancelLabel: "Cancel",
+            cancelLabel: "cancel",
             hideDialog: () => {
               this.setState({ openModal: null });
             }
           }
         }
       });
-    } finally {
-      opened = true;
+    } else {
+      // renaming
+      this.showDialog(ErrorDialog, {
+        title: "Name in Use",
+        message: "Node name is already in use. Please choose another.",
+        confirmLabel: "Okay"
+      });
```

Comment:
When does this happen? Maybe turn the `else` into an `else if` to make it explicit?